src/main/java/kitchenpos/eatinorders/domain/EatInOrder.java

```diff
@@ -0,0 +1,101 @@
+package kitchenpos.eatinorders.domain;
+
+import jakarta.persistence.*;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.UUID;
+
+@Table(name = "orders")
+@Entity
+public class EatInOrder {
+
+  @Column(name = "id", columnDefinition = "binary(16)")
+  @Id
+  private UUID id;
+
+  @Column(name = "type", nullable = false, columnDefinition = "varchar(255)")
+  @Enumerated(EnumType.STRING)
+  private EatInOrderType type;
+
+  @Column(name = "status", nullable = false, columnDefinition = "varchar(255)")
+  @Enumerated(EnumType.STRING)
+  private EatInOrderStatus status;
+
+  @Embedded private EatInOrderLineItems orderLineItems;
+  @Embedded private OrderDateTime orderDateTime;
+
+  @ManyToOne
+  @JoinColumn(
+      name = "order_table_id",
+      columnDefinition = "binary(16)",
+      foreignKey = @ForeignKey(name = "fk_orders_to_order_table"))
+  private EatInOrderTable orderTable;
+
+  protected EatInOrder() {}
+
+  protected EatInOrder(
+      final UUID id,
+      final List<EatInOrderLineItem> orderLineItems,
+      final EatInOrderTable orderTable) {
+    this.id = id;
+    this.type = EatInOrderType.EAT_IN;
+    this.status = EatInOrderStatus.WAITING;
+    this.orderLineItems = new EatInOrderLineItems(orderLineItems);
+    this.orderDateTime = new OrderDateTime(LocalDateTime.now());
+    this.orderTable = orderTable;
+  }
+
+  public static EatInOrder createOrder(
+      final List<EatInOrderLineItem> orderLineItems, final EatInOrderTable orderTable) {
+    return new EatInOrder(UUID.randomUUID(), orderLineItems, orderTable);
+  }
+
+  public void accepted() {
+    if (this.status != EatInOrderStatus.WAITING) {
+      throw new IllegalStateException();
```

Comment:
각 주문상태변경 메서드마다 반복되는 상태 체크로직을 아래 예시와 같은 공통 메서드로 추출하면
조금 더 가독성을 높혀 볼 수 있을 것 같아요
```suggestion
    checkOrThrow(EatInOrderStatus.WAITING);
```
```
    private void checkOrThrow(final OrderStatus status) {
        if (this.status != status) {
            throw new IllegalStateException("주문 상태가 " + status.name() + "가 아닙니다.");
        }
    }
```